src/client/src/components/register.js

```diff
@@ -0,0 +1,62 @@
+const React = require('react')
+const {Link, Redirect} = require('react-router-dom')
+const _fetch = require('node-fetch')
+
+const SUCCESS_CODE = 200
+const REGISTER_URL = '/register', HOMEPAGE_PATH = '/', POST = 'POST', headers = { 'Content-Type': 'application/json' }
+
+class Register extends React.Component {
+	constructor(props) {
+		super(props)
+
+		this.state = {
+			username: '',
+			password: '',
+			email: '',
+			successfulRegister: false
+		}
+		this.handleChange = this.handleChange.bind(this)
+		this.handleSubmit = this.handleSubmit.bind(this)
+	}
+	handleChange(event) {
+		this.setState({ [event.target.name] : event.target.value})
```

Comment:
What if server will send some strange event, not predicted by client?

Reply:
See above